mapbox_vector_tile/decoder.py

```diff
@@ -89,9 +90,18 @@ def parse_geometry(self, geom, ftype):
             i = i + 1
 
             if cmd == CMD_SEG_END:
-                break
+                if ftype == POLYGON and coords:
+                    coords.append(coords[0])
+                parts.append(coords)
+                coords = []
+
+            elif cmd == CMD_MOVE_TO or cmd == CMD_LINE_TO:
+
+                if ftype == LINESTRING and coords and cmd == CMD_MOVE_TO:
```

Comment:
The [polygon encoder code](https://github.com/mapzen/mapbox-vector-tile/blob/5fce2aa280f7184a19d01fa59d565fccbfd6d8dc/mapbox_vector_tile/encoder.py#L186-L187) has an explicit CMD_SEG_END command that gets inserted when a ring terminates. It doesn't do that for linestrings which is why I added this explicitly there.